pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -40,110 +51,193 @@ func (r *Condition) Update(other Condition) {
 	r.Type = other.Type
 	r.Status = other.Status
 	r.Reason = other.Reason
+	r.Category = other.Category
 	r.Message = other.Message
 	r.LastTransitionTime = metav1.NewTime(time.Now())
 }
 
 func (r Condition) Equal(other Condition) bool {
 	return r.Type == other.Type &&
 		r.Status == other.Status &&
+		r.Category == other.Category &&
 		r.Reason == other.Reason &&
 		r.Message == other.Message
 }
 
 // Managed collection of conditions.
 // Intended to be included in resource Status.
+// List - The list of conditions.
+// staged - In `staged` mode, the search methods like
```

Comment:
Refreshing my memory of why we have the concept of "staged" conditions... If I recall correctly, the crux is that it allows us to keep our old `lastTransitionTime` if nothing changes.

Going to keep reading and try to understand why we have the additional `BeginStagedConditions()` and `EndStagedConditions()` now.

Reply:
Not exactly.  I keeps the controller from having to explicitly deleting conditions that not longer exist.  The condition is marked as staged=true by SetCondition().  Then, EndStagedConditions() removes the conditions that were not marked as staged.